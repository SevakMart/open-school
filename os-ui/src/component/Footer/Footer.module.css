.mainContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 5% 2.7%;
  padding-top: 8%;
  padding-bottom: 8%;
  border-radius: 2.5rem;
  background-color: #52719E;
  box-sizing: border-box;
}
.mainContainer h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 800;
  font-size: 3rem;
  line-height: 3.5rem;
  color: #ffffff;
  margin-bottom: inherit;
}
.mainContainer .buttonContainer {
  width: 50%;
  text-align: center;
  box-sizing: border-box;
}

h2 {
  font-family: "Helvetica Neue";
  font-style: italic;
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.5rem;
  color: #222427;
  margin-left: 2.7%;
}/*# sourceMappingURL=Footer.module.css.map */