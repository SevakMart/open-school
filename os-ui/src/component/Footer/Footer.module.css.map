{"version":3,"sources":["Footer.module.scss","../../styles/Mixins/FlexMixins.scss","../../styles/Configs/Colors.scss","Footer.module.css","../../styles/Mixins/FontMixin.scss","../../styles/Configs/Fonts.scss"],"names":[],"mappings":"AAMA;ECLE,aAAA;EACA,sBDKmB;ECJnB,mBDI2B;ECH3B,uBDGmC;EACnC,eAAA;EACA,eAAA;EACA,kBAAA;EACA,qBAAA;EACA,yBEZc;EFad,sBAAA;AGFF;AHGI;EIbF,yBCDY;EDEZ,kBJasC;EIZtC,gBJY8C;EIX9C,eJWmD;EIVnD,mBJUyD;EACrD,cEZa;EFab,sBAAA;AGGN;AHAE;EACE,UAAA;EACA,kBAAA;EACA,sBAAA;AGEJ;;AHCA;EIzBE,6BAAA;EACA,kBJyBoC;EIxBpC,gBJwB4C;EIvB5C,kBJuBiD;EItBjD,mBJsB0D;EAC1D,cEzBkB;EF0BlB,iBAAA;AGMF","file":"Footer.module.css"}