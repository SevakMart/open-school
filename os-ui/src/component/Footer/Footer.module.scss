@import "../../styles/Mixins/FlexMixins.scss";
@import "../../styles/Mixins/FontMixin.scss";
@import "../../styles/Configs/Colors.scss";
@import "../../styles/Configs/Fonts.scss";

.mainContainer {
  @include flexStyle(column, center, center);
  margin: 5% 2.7% 7%;
  padding-top: 45px;
  padding-bottom: 6%;
  border-radius: 2.5rem;
  background-color: $primary-color;
  h2 {
    @include fontStyle($primary-font, normal, 800, 3rem, 3.5rem);
    color: $white-text-color;
    margin-bottom:inherit;
  }
  .buttonContainer {
    padding-top: 40px;
    width: 50%;
    text-align: center;
  }
}

@media only screen and (max-width: 1187px) {
  .mainContainer {
   .buttonContainer {
    button {
      margin-top: 18px;
    }
   } 
  }
}

@media only screen and (max-width: 775px) {
  .mainContainer {
    h2 {
      font-weight: 500;
      font-size: 2.5rem;
      line-height: 3rem;
    }
  }
}

@media only screen and (max-width: 610px) {
  .mainContainer {
    h2 {
      font-weight: 400;
      font-size: 2rem;
      line-height: 2rem;
    }
    .buttonContainer {
      padding-top: 40px;
      width: 80%;
    }
  }
}

@media only screen and (max-width: 480px) {
  .mainContainer {
    h2 {
      font-weight: 300;
      font-size: 1.4rem;
      line-height: 1rem;
    }
  }
}