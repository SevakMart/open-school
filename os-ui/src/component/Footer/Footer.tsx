import { useTranslation } from 'react-i18next';
import styles from './Footer.module.scss';

const Footer = () => {
  const { t } = useTranslation();
  const {
    mainContainer, categories, becomeMentor, findUsOn,
  } = styles;
  return (
    <div className={mainContainer}>
      <button type="button">{t('string.homePage.navBar.logo')}</button>
      <div className={categories}>
        <h2>{t('string.homePage.footer.categories')}</h2>
        <p>{t('string.homePage.footer.softwareEngineering')}</p>
        <p>{t('string.homePage.footer.design')}</p>
        <p>{t('string.homePage.footer.management')}</p>
      </div>
      <div className={becomeMentor}>
        <h2>{t('string.homePage.footer.becomeMentor')}</h2>
        <p>{t('string.homePage.footer.about')}</p>
        <p>{t('string.homePage.footer.help')}</p>
      </div>
      <div className={findUsOn}>
        <h2>{t('string.homePage.footer.findUs')}</h2>
        <p>Facebook</p>
        <p>Instagram</p>
      </div>
    </div>
  );
};
export default Footer;
