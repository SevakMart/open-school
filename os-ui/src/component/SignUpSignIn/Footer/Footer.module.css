.alreadyHaveAccount {
  text-align: center;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #5E617B;
  margin-top: 24px;
}/*# sourceMappingURL=Footer.module.css.map */