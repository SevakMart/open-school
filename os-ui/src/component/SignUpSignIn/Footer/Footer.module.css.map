{"version":3,"sources":["Footer.module.scss","../../../styles/Mixins/FontMixin.scss","../../../styles/Configs/Fonts.scss","../../../styles/Configs/Colors.scss","Footer.module.css"],"names":[],"mappings":"AAIA;EACI,kBAAA;ECJF,yBCDY;EDEZ,kBDIoC;ECHpC,gBDG4C;ECF5C,eDEiD;ECDjD,mBDCuD;EACrD,cGLc;EHMd,gBAAA;AICJ","file":"Footer.module.css"}