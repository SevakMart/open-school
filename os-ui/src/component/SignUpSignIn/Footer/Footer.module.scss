@import "../../../styles/Configs/Colors.scss";
@import "../../../styles/Configs/Fonts.scss";
@import "../../../styles/Mixins/FontMixin.scss";

.alreadyHaveAccount {
    text-align: center;
    @include fontStyle($primary-font, normal, 500, 1rem, 1.5rem);
    color: $main-text-color;
    margin-top:24px;
}