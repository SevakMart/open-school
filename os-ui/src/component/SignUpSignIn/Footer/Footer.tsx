import { useDispatch } from 'react-redux';
import Button from '../../Button/Button';
import { Types } from '../../../types/types';
import { openModal } from '../../../redux/Slices/PortalOpenStatus';
import styles from './Footer.module.scss';

/* eslint-disable max-len */
const Footer = ({ mainText, buttonType, buttonText }:{mainText:string, buttonType:string, buttonText:string}) => {
  const { alreadyHaveAccount } = styles;
  const dispatch = useDispatch();

  const handleSignIn = () => {
    dispatch(openModal({ buttonType }));
  };

  const handleSignUp = () => {
    dispatch(openModal({ buttonType }));
  };

  return (
    <>
      <div className={alreadyHaveAccount}>
        {mainText}
        <span>
          {buttonType === Types.Button.SIGN_IN && <Button.SignInButton className={['account']} onClick={handleSignIn}>{buttonText}</Button.SignInButton>}
          {buttonType === Types.Button.SIGN_UP && <Button.SignUpButton className={['account']} onClick={handleSignUp}>{buttonText}</Button.SignUpButton>}
        </span>
      </div>
    </>
  );
};
export default Footer;
