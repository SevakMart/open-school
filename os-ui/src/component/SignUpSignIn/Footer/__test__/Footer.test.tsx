import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../../../redux/Store';
import Footer from '../Footer';

describe('Create unit tests for Footer component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Footer
          mainText="This is main text"
          buttonType="signIn"
          buttonText="Sign In"
        />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
