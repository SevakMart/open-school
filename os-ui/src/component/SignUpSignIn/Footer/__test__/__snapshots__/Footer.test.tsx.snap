// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create unit tests for Footer component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="alreadyHaveAccount"
  >
    This is main text
    <span>
      <button
        class="account"
        type="button"
      >
        Sign In
      </button>
    </span>
  </div>
</DocumentFragment>
`;
